src/Common/Core/Test/Fixtures/CoreShellProviderFixture.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.ComponentModel.Composition.Hosting;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Extensions;
+using Microsoft.Common.Core.Services;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.Test.Fakes.Shell;
+using Microsoft.UnitTests.Core.Shell;
+using Microsoft.UnitTests.Core.XUnit;
+using Microsoft.UnitTests.Core.XUnit.MethodFixtures;
+using Xunit.Sdk;
+
+namespace Microsoft.Common.Core.Test.Fixtures {
+    public abstract class CoreShellProviderFixture : MethodFixtureBase, ICoreShellProvider {
```

Comment:
Tests are unified to use core shell (as the rest of the code). So it is simpler to have shell fixture rather than one method or providing services there and another here. Get core shell, get stuff from it. 

Same for the rest of the code (and in the upcoming editor split). I mostly pass core shell around, makes it a lot simpler that passing multiple separate parameters or service container. Makes it easier to make tactical fixes in the end phase too. Adding service and consuming it makes 2 line change, plumbing new argument through 20 nested calls is much harder. I already had to hack fix for Win32Exception where there is no core shell - was not possible to show user message/prompt.